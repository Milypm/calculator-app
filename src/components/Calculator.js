import React from 'react';
import Navbar from './Navbar';
import App from './App';
import '../style/style.css';

const Calculator = () => (
  <div className="first-div calc-first-div">
    <Navbar />
    <div className="calc-div">
      <h2 className="calc-h2">Let&apos;s do some math!</h2>
      <App />
    </div>
  </div>
);

export default Calculator;
