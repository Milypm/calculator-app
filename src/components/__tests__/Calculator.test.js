import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Calculator from '../Calculator';

describe('Calculator component', () => {
  test('renders Calculator component', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Calculator />
      </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
