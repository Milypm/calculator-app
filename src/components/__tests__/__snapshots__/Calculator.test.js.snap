// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component displays specific text on the screen 1`] = `
<div
  className="first-div calc-first-div"
>
  <div
    className="header-navbar"
  >
    <div
      className="header-icon"
    >
      <h1>
        Math Magicians
      </h1>
      <i
        className="fas fa-hat-wizard"
      />
    </div>
    <ul
      className="nav-ul"
    >
      <li
        className="li-separator"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="li-separator li-calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote-of-the-day"
          onClick={[Function]}
        >
          Quote of the Day
        </a>
      </li>
    </ul>
  </div>
  <div
    className="calc-div"
  >
    <h2
      className="calc-h2"
    >
      Let's do some math!
    </h2>
    <div>
      <div
        className="calc-container display-container"
      >
        <p>
          0
        </p>
      </div>
      <div
        className="calc-container btnPanel-container"
      >
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              AC
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              +/-
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              %
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              รท
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              7
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              8
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              9
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              x
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              4
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              5
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              6
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              -
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              1
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              2
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              3
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              +
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey large"
            onClick={[Function]}
            type="button"
          >
            <span>
              0
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              .
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              =
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Calculator component renders Calculator component 1`] = `
<div
  className="first-div calc-first-div"
>
  <div
    className="header-navbar"
  >
    <div
      className="header-icon"
    >
      <h1>
        Math Magicians
      </h1>
      <i
        className="fas fa-hat-wizard"
      />
    </div>
    <ul
      className="nav-ul"
    >
      <li
        className="li-separator"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="li-separator li-calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote-of-the-day"
          onClick={[Function]}
        >
          Quote of the Day
        </a>
      </li>
    </ul>
  </div>
  <div
    className="calc-div"
  >
    <h2
      className="calc-h2"
    >
      Let's do some math!
    </h2>
    <div>
      <div
        className="calc-container display-container"
      >
        <p>
          0
        </p>
      </div>
      <div
        className="calc-container btnPanel-container"
      >
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              AC
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              +/-
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              %
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              รท
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              7
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              8
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              9
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              x
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              4
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              5
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              6
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              -
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              1
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              2
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              3
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              +
            </span>
          </button>
        </div>
        <div
          className="btns-row"
        >
          <button
            className="button-container grey large"
            onClick={[Function]}
            type="button"
          >
            <span>
              0
            </span>
          </button>
          <button
            className="button-container grey normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              .
            </span>
          </button>
          <button
            className="button-container orange normal"
            onClick={[Function]}
            type="button"
          >
            <span>
              =
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
